client/rpc_sender.go

```diff
@@ -83,6 +84,13 @@ func newRPCSender(server string, context *base.Context, retryOpts retry.Options,
 // and been executed successfully. We retry here to eventually get through with
 // the same client command ID and be given the cached response.
 func (s *rpcSender) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.BatchResponse, *roachpb.Error) {
+	timeout := 5 * s.retryOpts.MaxBackoff
```

Comment:
> This patch effectively changes the connect timeout from 250ms to 25s, which seems excessive. It would be more appropriate to change the InitialBackoff in use.

I don't think that's true; this patch is more precise than that, it increases the connect timeout on just the very first connection. I agree the actual number is excessive, I'll change that. Increasing `InitialBackoff` will have an effect on every single request, not just the first one which occurs before the client is connected.